cleverhans/model_abs.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta
+from keras.models import Model
+
+
+class ModelAbstraction:
+    """
+    An abstract interface for a model that exposes the required functionalities
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model):
+        """
+        :param model: A function that takes a symbolic input and returns the
+                      symbolic output for the model's predictions.
+        """
+        raise NotImplementedError('Model abstraction not implementated.')
+
+    def get_hidden(self, layer):
```

Comment:
Right. I had only Keras in mind when thinking about the overhead. But still I'm not sure returning all hidden layers at the same time is needed. Feature adversaries attack one layer at a time.

Let's make sure we agree on how the model abstraction is going to be used in the group. I send an email.